### PR TITLE
Fixed internal error when adding an alias

### PR DESCRIPTION
**Pull request by mchf**
bnc#864264

### CODE REVIEW

**Comment by jreidinger** [comment]
LGTM

**Comment by mvidner** [comment]
Actually this does not fix the originally reported bug. It fixes a bug that we introduced only 2 months ago in https://github.com/yast/yast-network/commit/4efb006a22ee0dd8489ffdc023c099670d5a09da#diff-4b8e1b8810d82526b81cb37dcf76f811R244
```ruby
-      while true
+      loop do
         host = nil
         ret = UI.UserInput
```

(and that should warn us to be careful when "fixing" Rubocop problems)

**Comment by mvidner** [comment]
Moving a `CloseDialog` into a loop looks suspicious, and indeed it will break if the user presses Cancel. IMHO a better fix is to declare `ret = nil` before the `loop`.

Why don't we add unit tests for the dialog? It should be possible now thanks to Ancor's work in journald.

**Comment by mvidner** [comment]
The code looks good now, but we should add tests. Let me first try resurrecting the coverage PR #307 so that we can see an improvement in the metric ;-)

File: test/alias_dialog_test.rb [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+      .with(Id(:ipaddr), :Value)
+      .and_return("1.1.1.1")
+    allow(Yast::UI)
+      .to receive(:QueryWidget)
+      .with(Id(:netmask), :Value)
+      .and_return("255.0.0.0")
+    allow(Yast::LanItems).to receive(:device).and_return("")
+
+    expect(Yast::UI).to receive(:UserInput).once
+    expect(AliasDialog.new.VirtualEditDialog(0, term(:empty), nil)).not_to be nil
```
Comment (mvidner) [inline test/alias_dialog_test.rb]: `0, term(:empty), nil` ? Even if this is only a smoke test, I have absolutely no idea what the arguments mean. I'll write an improvement later today.

Comment (mchf) [inline test/alias_dialog_test.rb]: that's ... bad design ... I can tell you what is what ... but ... you'll see ;)

**Comment by mvidner** [comment]
OK, in the end I just removed the disused argument and documented the method API. Please add https://github.com/mvidner/yast-network/commit/3cecd09f723f66fb36b3cacf32041b5f28a6042b from https://github.com/mvidner/yast-network/commits/master , then it LGTM.

**Comment by mchf** [comment]
updated according instructions ;-)

Going to merge once travis is done. Thanks for reviews.

**Pull request merged**